src/smart_building_rating_calculator/main.py

```diff
@@ -0,0 +1,76 @@
+from src.smart_building_rating_calculator.inputs import (
+    UserInputs,
+    EVChargerPower,
+    BatterySize,
+    SolarInverterSize,
+    HeatingSource,
+    HotWaterSource,
+)
+from src.smart_building_rating_calculator.scoring import (
+    calc_electrification_score,
+    calc_ics_score,
+)
+from src.smart_building_rating_calculator.flex_archetype import calc_flex_archetype
+
+
+def calc_sbr(sbr_val: float) -> str:
+    if sbr_val > 22:
+        return "A"
+    elif sbr_val > 15:
+        return "B"
+    elif sbr_val > 10:
+        return "C"
+    elif sbr_val > 6:
+        return "D"
+    elif sbr_val > 4:
+        return "E"
+    elif sbr_val > 1:
+        return "F"
+    else:
+        return "G"
+
+
+def sbr_score(
+    smart_meter: bool,
+    smart_ev_charger: bool,
+    charger_power: EVChargerPower,
+    smart_v2g_enabled: bool,
+    home_battery: bool,
+    battery_size: BatterySize,
+    solar_pv: bool,
+    pv_inverter_size: SolarInverterSize,
+    electric_heating: bool,
+    heating_source: HeatingSource,
+    hot_water_source: HotWaterSource,
+    secondary_heating: bool,
+    secondary_hot_water: bool,
+    integrated_control_sys: bool,
+):
```

Comment:
Return typing:

```python

from typing import Tuple

def test_fun(x: int) -> Tuple[int, int, int]:
    return 1,2,3
```

Something like that - can't remember exact syntax but that's how you can type multiple returns.

Reply:
Also have a think about the user experience with this function. For someone that has pip-installed this library, what's their experience like calling this function? At the moment, they have to run _a lot_ of import statements just to be able to call this function.

It'd be much easier for this function to take `UserInput` class the input. Have a separate function that takes in 'user inputs' to instantiate the `UserInput` class. Also have a think of how user will be creating the inputs.

For web clients, these would likely be in a JSON object/ string/ int values. A wrapper function to convert this string inputs into UserInput object which is then parsed into this function would be more convenient for end users.

You can then also write unit tests to test that inputs are parsed properly.

Implementation could be very similar to Gus's feedback on "instantiate the UserInputs dataclass for each flex archetype and then check for equality".

Reply:
I would also avoid naming this script `main.py`.

`main.py` or `app.py` is usually reserved for an application where `main.py`/`app.py` are the entry points to said app.

I would call this script "calculate_sbr_score.py`.

Think about the common python libraries that you use - I don't think you'll have ever run `from package.main import fn`!

Reply:
Renamed the script to `calculate_sbr_score.py` as recommended.